tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/ticket/TicketManager.scala

```diff
@@ -29,10 +32,23 @@ case class Ticket(clientAddress: InetAddress,
                   username: Username)
 
 object TicketManager {
-  val validity: java.time.Duration = java.time.Duration.ofMinutes(1)
+  private val validity: java.time.Duration = java.time.Duration.ofMinutes(1)
+  private val ipValidationEnableProperty: String = "authentication.strategy.rfc8621.tickets.ip.validation.enabled"
+  private val ipValidationEnabled: Boolean = true
+
+  def extractIpValidationEnableConfig(propertiesProvider: PropertiesProvider): Boolean =
+    try {
+      Option(propertiesProvider.getConfiguration("jmap"))
+        .map(config => config.getBoolean(ipValidationEnableProperty, ipValidationEnabled))
+        .getOrElse(ipValidationEnabled)
+    } catch {
+      case _: FileNotFoundException => ipValidationEnabled
+    }
```

Comment:
Why not put that into JMAPExtensionConfiguration ?

Over places where we manually acces the JMAP conf file:

```
XXX:~/Documents/openpaas-james/tmail-backend/jmap/extensions$ grep 'getConfiguration("jmap")' * -r
src/main/scala/com/linagora/tmail/james/jmap/method/CalendarEventReplyPerformer.scala:    Try(propertiesProvider.getConfiguration("jmap"))
src/main/scala/com/linagora/tmail/james/jmap/method/EmailRecoveryActionSetMethod.scala:    val config = Try(propertiesProvider.getConfiguration("jmap"))
src/main/scala/com/linagora/tmail/james/jmap/oidc/WebFingerRoutes.scala:    Try(propertiesProvider.getConfiguration("jmap"))
```
Time for a refactoring?

Reply:
I just see that. Let's refactor a bit there.